src/components/timeline/TrackScreenshots.js

```diff
@@ -98,21 +98,32 @@ class Screenshots extends PureComponent<Props, State> {
     });
   };
 
-  _handleClick = (event: SyntheticMouseEvent<HTMLDivElement>) => {
-    const { screenshots, updatePreviewSelection } = this.props;
+  // This selects a screenshot when clicking on the screenshot strip. Note that
+  // we use the mouseup event so that isMakingPreviewSelection is still
+  // accurate.
+  _selectScreenshotOnClick = (event: SyntheticMouseEvent<HTMLDivElement>) => {
+    const {
+      screenshots,
+      updatePreviewSelection,
+      isMakingPreviewSelection,
+    } = this.props;
+    if (isMakingPreviewSelection) {
+      // Avoid reseting the selection if the user is currently selecting one.
+      return;
+    }
+
     const { left } = event.currentTarget.getBoundingClientRect();
     const offsetX = event.pageX - left;
     const screenshotIndex = this.findScreenshotAtMouse(offsetX);
-    if (screenshotIndex === null) return null;
+    if (screenshotIndex === null) return;
```

Comment:
Note: I was surprised that these if blocks don't have brackets. I'll add the "curly" rule to eslint soon.

Reply:
Hah, wow yeah I was about to comment on this.